### ISSUE TITLE
sdist is missing tests

### ISSUE THREAD

**Issue by mtelka** [encode/httpcore#798]
- [x ] Initially raised as discussion #796

The sdist package version 0.18.0 at PyPI is missing tests. This is regression from version 0.17.3.  Please add tests back to sdist.  Thank you.

**Comment by karosis88** [comment]
@mtelka Would you like to issue a PR resolving this?

**Comment by karosis88** [comment]
> > This is regression from version 0.17.3.
> 
> How can I observe that this is a regression?

This issue comes from #752. In 0.17.3 we have tests in sdist.

**Comment by mtelka** [comment]
> > This is regression from version 0.17.3.
> 
> How can I observe that this is a regression?

```
$ for v in 0.17.3 0.18.0 ; do
    wget --quiet https://pypi.io/packages/source/h/httpcore/httpcore-$v.tar.gz
    tar xf httpcore-$v.tar.gz
    [ -d httpcore-$v/tests ] && echo "$v: tests in sdist" || echo "$v: tests are missing"
done
0.17.3: tests in sdist
0.18.0: tests are missing
$
```

**Comment by mtelka** [comment]
> @mtelka Would you like to issue a PR resolving this?

Sorry, no. ENOTIME.

**Comment by mtelka** [comment]
It would be also great to include [requirements.txt](https://github.com/encode/httpcore/blob/master/requirements.txt) in sdist so we can automatically detect testing dependencies before we start tests.

Also, would you mind to add minimal `tox.ini`?  Something similar to [this one](https://github.com/typeddjango/pytest-mypy-plugins/pull/121)?

Thank you.

**Comment by tomchristie** [comment]
I've closed this issue due to lack of respect.

**Comment by tomchristie** [comment]
My response was unnecessarily brusque, here's what I could have said;

Following an "Sorry, no. ENOTIME." immediately with further requests for us to change our tooling is going to ellict a non-constructive response from me. It's disrespectful to the time & effort our team put into this project.

---

Getting back onto a collaborative track...

If you'd like to request either of those two tooling changes, I'd suggest opening a separate discussion for them.

**Comment by mtelka** [comment]
Okay, @tomchristie  you are right.  Sorry for that.  With ENOTIME I meant that there are a lot of build backends (hatch, poetry, flit, setuptools, ...) in the wild and all of them use a bit different way how to add things to sdist.  Sometimes it is via `MANIFEST.in`, sometimes using some config setup in `pyproject.toml`, etc.

The actual (default) content of sdist even depends on tool versions used to create sdist.  For example I faced a case when a project used some older version of [build](https://pypi.org/project/build/) to create their sdists and so there were some file(s) missing in sdist.  When I tried to create sdist for such a project using the latest version of [build](https://pypi.org/project/build/) the sdist was complete.  So the proper way how to fix missing file in sdist was just to upgrade [build](https://pypi.org/project/build/) on the project's side (no code change was needed), IIRC.  Or the difference was caused by some other tool in the chain?  I do not remember :-).

This is something that is far better handled by developers of the project.  It is usually trivial one-line change.  The only problem is to find where the change needs to be done.  A random bystander like me would need to spend far more time to find proper place for such a change than project developer.  Thanks @karosis88 for the PR!

Again, I'm sorry for misunderstanding.

OTOH, now when I see how you are going to add tests to sdist I know how to deal with this stuff for `httpcore`, so I can create PR with either `tox.ini` or `requirements.txt`.
